capsicum-test-main.cc

```diff
@@ -6,10 +6,47 @@
 #include <ctype.h>
 #include <errno.h>
 #include <pwd.h>
+#include <stdio.h>
+#include <stdlib.h>
 #include <iostream>
 #include "gtest/gtest.h"
 #include "capsicum-test.h"
 
+std::string tmpdir;
+
+class SetupEnvironment : public ::testing::Environment
+{
+public:
+  SetupEnvironment() : teardown_tmpdir_(false) {}
+  void SetUp() override {
+    if (tmpdir.empty()) {
+      std::cerr << "Generating temporary directory root: ";
+      CreateTemporaryRoot();
+    } else {
+      std::cerr << "User provided temporary directory root: ";
+    }
+    std::cerr << tmpdir << std::endl;
+  }
+  void CreateTemporaryRoot() {
+    char *tmp_root;
+    char *tmp_root_template = tempnam(nullptr, "cptst");
+
+    ASSERT_NE(tmp_root_template, nullptr);
+    tmp_root = mkdtemp(tmp_root_template);
```

Comment:
This still fails, I think because Linux `mkdtemp` [requires](http://man7.org/linux/man-pages/man3/mkdtemp.3.html) that "The last six characters of template must be XXXXXX".

Maybe just `mkdir` ?

Reply:
@daviddrysdale: good call; done!